core/rollup_l1_cost.go

```diff
@@ -43,25 +41,18 @@ var (
 // It returns nil if there is no applicable cost function.
 func NewL1CostFunc(config *params.ChainConfig, statedb vm.StateDB) vm.L1CostFunc {
```

Comment:
Do we still need to cache if we know that the semantics now are "read the values from state after applying the deposits" - technically the semantics are "read the values from state after applying the system tx" but after the deposits feels a bit better to me. The tradeoff is that you don't know the number of deposits beforehand

Reply:
I guess what I'm getting at here is 2 things.

- Defining the exact semantics so that we can place them in the spec
- Potentially reducing the diff

Reply:
We may need the values for multiple transactions per block, so the cache is still useful I think